app/views/report/_widget_show.html.haml

```diff
@@ -1,151 +1 @@
-    - elsif @sb[:wtype] == "c"
-      .form-group
-        %label.control-label.col-md-2
-          - if @widget.read_only
-            *
-          = _('Filter')
-        .col-md-10
-          %p.form-control-static
-            - if @widget.resource
-              = h(@widget.resource.title)
-            - else
-              %span{:style => "color: red;"}
-                = "*** %s ***" % _("Chart no longer exists")
-    - elsif @sb[:wtype] == "m"
-      .form-group
-        %label.control-label.col-md-2
-          - if @widget.read_only
-            *
-          = _('Shortcuts')
-        .col-md-10
-          %p.form-control-static
-            - if true
-              -# TODO: fix formatting to look like widget menu shortcuts (first attempt in ELSE)
-              - @widget.miq_widget_shortcuts.order("sequence").each_with_index do |ws, wsi|
-                - unless wsi == 0
-                  %br
-                = ws.description
-            - else
-              .row#modules
-                .col-md-4#col1
-                  - @widget.miq_widget_shortcuts.each do |ws|
-                    %div
-                      .panel.panel-default
-                        .panel-header
-                          %h3.panel-title
-                            %a{:class => "#{@widget.content_type}box"}
-                            = ws.description
-    - if @sb[:wtype] == "r"
-      - @sb[:col_order].each_with_index do |col, idx|
-        .form-group
-          %label.control-label.col-md-2= _("Column %{index}") % {:index => (idx + 1)}
-          .col-md-10
-            %p.form-control-static= h(col)
-
-    - if %w(r rf).include?(@sb[:wtype])
-      .form-group
-        %label.control-label.col-md-2= _('Row Count')
-        .col-md-10
-          %p.form-control-static= h(@widget.row_count)
-
-  %hr
-  - unless @sb[:wtype] == "m"
-    %h3= _('Timer')
-    .form-horizontal
-      - if s.kind_of?(MiqSchedule)
-        .form-group
-          %label.control-label.col-md-2= _('Run At')
-          .col-md-10
-            %p.form-control-static= h(s.run_at_to_human(@timezone).to_s)
-        .form-group
-          %label.control-label.col-md-2= _('Next Run Time')
-          .col-md-10
-            %p.form-control-static
-              - unless s.next_run_on.blank?
-                = h(format_timezone(s.next_run_on, tz, "view"))
-      - else
-        .form-group
-          %label.control-label.col-md-2
-          .col-md-10
-            %p.form-control-static
-              %b= _('No timer is attached to this Widget, its contents will not be updated.')
-              %br= _('Edit this Widget to configure a timer.')
-    %hr
-  %h3= _('Visibility')
-  .form-horizontal
-    .form-group
-      %label.control-label.col-md-2= _('Show')
-      .col-md-10
-        %p.form-control-static= visibility_options(@widget)
-  - if @widget.read_only
-    = _('* Fields are read only for default Widgets')
+= dashboard_widget_summary(@widget, @sb)
```

Comment:
Minor (and it can be done in a followup), but you can get rid of the haml altogether and switch to erb.  Ideally, we should get rid of haml to have one less dependency.

To switch you can do:

- Rename from _widget_show.html.haml -> _widget_show.html.erb
- Make the content:

```erb
<%= dashboard_widget_summary(@widget, @sb) %>
```